### PR TITLE
Added Jenkinsfile to automatically build benchmarks and run tests.

### PR DESCRIPTION
**Pull request by rpgolshan**
Jenkinsfile is a pipeline flow for a multibranch pipeline project. This
commit adds stages to build every main directory (e.g. lockhammer) and
run all tests starting with directory/scripts/test*

Jenkins will update any pull requests statuses with either a 'success'
or a 'failure' while listing what failed.

### CODE REVIEW

**Comment by geoffreyblake** [comment]
Can one of the admins verify this patch?

**Comment by geoffreyblake** [comment]
Can one of the admins verify this patch?

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        setBuildStatus("Testing code", 'PENDING', 'Test');
+
+        fails = []
+        dir = new File("${env.WORKSPACE}/benchmarks/");
+        // Run all scripts starting with the prefix "test"
+        dir.traverse(type: DIRECTORIES, maxDepth: 0) {
+            def scr = new File("${it}/scripts/")
+            scr.traverse(type: FILES, filter: ~/.*\/test[^\/]*/) {
+                try {
+                    sh it
```
Comment (geoffreyblake) [inline Jenkinsfile]: Is this expecting tests are shell scripts, or does this work for an executable python script as well?

Comment (rpgolshan) [reply Jenkinsfile]: It should run "it" in a bourne shell. So if the test script is executable and it has a shebang at the top, it should run correctly.

**Pull request merged**